test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+        },
+        {
+            buf = '21-08-2023 Monday',
+            fmt = '%d-%m-%Y %A',
+        },
+        {
+            buf = '22-08-2023 Tuesday',
+            fmt = '%d-%m-%Y %A',
+        },
+        {
+            buf = '23-08-2023 Wednesday',
+            fmt = '%d-%m-%Y %A',
+        },
+        {
+            buf = '24-08-2023 Thursday',
+            fmt = '%d-%m-%Y %A',
+        },
+        {
+            buf = '25-08-2023 Friday',
+            fmt = '%d-%m-%Y %A',
+        },
+        {
+            buf = '26-08-2023 Saturday',
+            fmt = '%d-%m-%Y %A',
+        },
+        -- %b
+        {
+            buf = 'Jan',
+            fmt = '%b',
+        },
+        {
+            buf = 'Feb',
+            fmt = '%b',
+        },
+        {
+            buf = 'Mar',
+            fmt = '%b',
+        },
+        {
+            buf = 'Apr',
+            fmt = '%b',
+        },
+        {
+            buf = 'May',
+            fmt = '%b',
+        },
+        {
+            buf = 'Jun',
+            fmt = '%b',
+        },
+        {
+            buf = 'Jul',
+            fmt = '%b',
+        },
+        {
+            buf = 'Aug',
+            fmt = '%b',
+        },
+        {
+            buf = 'Sep',
+            fmt = '%b',
+        },
+        {
+            buf = 'Oct',
+            fmt = '%b',
+        },
+        {
+            buf = 'Nov',
+            fmt = '%b',
+        },
+        {
+            buf = 'Dec',
+            fmt = '%b',
+        },
+        -- %B
+        {
+            buf = 'January',
+            fmt = '%B',
+        },
+        {
+            buf = 'February',
+            fmt = '%B',
+        },
+        {
+            buf = 'March',
+            fmt = '%B',
+        },
+        {
+            buf = 'April',
+            fmt = '%B',
+        },
+        {
+            buf = 'May',
+            fmt = '%B',
+        },
+        {
+            buf = 'June',
+            fmt = '%B',
+        },
+        {
+            buf = 'July',
+            fmt = '%B',
+        },
+        {
+            buf = 'August',
+            fmt = '%B',
+        },
+        {
+            buf = 'September',
+            fmt = '%B',
+        },
+        {
+            buf = 'October',
+            fmt = '%B',
+        },
+        {
+            buf = 'November',
+            fmt = '%B',
+        },
+        {
+            buf = 'December',
+            fmt = '%B',
+        },
+        -- %c
+        {
+            buf = 'Thu Jan  1 03:00:00 1970',
+            fmt = '%c',
+        },
+        -- %C
+        {
+            buf = '26-08-2024 20',
+            fmt = '%d-%m-%Y %C',
+        },
+        -- %d
+        {
+            buf = '01',
+            fmt = '%d',
+        },
+        {
+            buf = '31',
+            fmt = '%d',
+        },
+        -- %D
+        {
+            buf = '09/09/04',
+            fmt = '%D',
+        },
+        -- %e
+        {
+            buf = ' 30',
```

Comment:
Do we need the whitespace here in the buffer? If yes, please add the corresponding comment.

Reply:
> Do we need the whitespace here in the buffer?

Yes: 

>  %e is replaced by the day of the month as a decima number (1-31); single digits are preceded by a blank.

> If yes, please add the corresponding comment.

Added:

```diff
diff --git a/test/app-tap/datetime.test.lua b/test/app-tap/datetime.test.lua
index abd09f8295..41c7cf1456 100755
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -1839,8 +1839,9 @@ end)
 --
 test:test("Parse strptime format and check with :format()", function(test)
     local formats = {
-        -- %a
-        -- Works for 'Thu' only when date is not complete.
+        -- %a is replaced by national representation of the
+        -- abbreviated weekday name.
+        -- Note, works for 'Thu' only when date is not complete.
         {
             buf = 'Thu',
             fmt = '%a',
@@ -1873,12 +1874,10 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '25-08-2023 Fri',
             fmt = '%d-%m-%Y %a',
         },
-        {
-            buf = '26-08-2023 Sat',
-            fmt = '%d-%m-%Y %a',
-        },
-        -- %A
-        -- Works for 'Thursday' only when date is not complete.
+        -- %A is replaced by national representation of the full
+        -- weekday name.
+        -- Note, works for 'Thursday' only when date is not
+        -- complete.
         {
             buf = 'Thursday',
             fmt  = '%A',
@@ -1911,11 +1910,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '25-08-2023 Friday',
             fmt = '%d-%m-%Y %A',
         },
-        {
-            buf = '26-08-2023 Saturday',
-            fmt = '%d-%m-%Y %A',
-        },
-        -- %b
+        -- %b is replaced by national representation of the
+        -- abbreviated month name.
         {
             buf = 'Jan',
             fmt = '%b',
@@ -1964,7 +1960,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = 'Dec',
             fmt = '%b',
         },
-        -- %B
+        -- %B is replaced by national representation of the full
+        -- month name.
         {
             buf = 'January',
             fmt = '%B',
@@ -2013,17 +2010,20 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = 'December',
             fmt = '%B',
         },
-        -- %c
+        -- %c is replaced by national representation of time and
+        -- date.
         {
             buf = 'Thu Jan  1 03:00:00 1970',
             fmt = '%c',
         },
-        -- %C
+        -- %C is replaced by (year / 100) as decimal number;
+        -- single digits are preceded by a zero.
         {
             buf = '26-08-2024 20',
             fmt = '%d-%m-%Y %C',
         },
-        -- %d
+        -- %d is replaced by the day of the month as a decimal
+        -- number (01-31).
         {
             buf = '01',
             fmt = '%d',
@@ -2032,18 +2032,22 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '31',
             fmt = '%d',
         },
-        -- %D
+        -- %D is equivalent to "%m/%d/%y".
         {
             buf = '09/09/04',
             fmt = '%D',
         },
-        -- %e
+        -- %e is replaced by the day of the month as adecima
+        -- number (1-31); single digits are preceded by a blank.
         {
             buf = ' 30',
             fmt = '%e',
         },
-        -- %E, modifier, skipped.
-        -- %F
+        -- %E is a POSIX locale extensions. The sequences %Ec %EC
+        -- %Ex %EX  %Ey %EY %Od %Oe %OH %OI %Om %OM %OS %Ou %OU
+        -- %OV %Ow %OW %Oy are supposed to provide alternate
+        -- representations.
+        -- %F is equivalent to "%Y-%m-%d".
         {
             buf = '1970-01-01T03:00:00.125',
             fmt = '%FT%T.%f',
@@ -2060,17 +2064,22 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '1970-01-01',
             fmt = '%F',
         },
-        -- %f, Tarantool-specific.
+        -- %f is replaced by the fractional seconds,
+        -- Tarantool-specific.
         {
             buf = '100',
             fmt = '%f',
         },
-        -- %G
+        -- %G is replaced by a year as a decimal number with
+        -- century. This year is the one that contains the greater
+        -- part of the week (Monday as the first day of the week).
         {
             buf = '2024',
             fmt = '%G',
         },
-        -- %g, it is not possible to roundtrip, skipped.
+        -- %g is replaced by the same year as in "%G", but as
+        -- a decimal number without century (00-99).
+        -- Skipped, it is not possible to roundtrip.
         -- %h, same as %b
         {
             buf = 'Jan',
@@ -2120,7 +2129,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = 'Dec',
             fmt = '%b',
         },
-        -- %H
+        -- %H, is replaced by the hour (24-hour clock) as
+        -- a decimal number (00-23).
         {
             buf = '00',
             fmt = '%H',
@@ -2129,7 +2139,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '23',
             fmt = '%H',
         },
-        -- %I
+        -- %I is replaced by the hour (12-hour clock) as
+        -- a decimal number (01-12).
         {
             buf = '01',
             fmt = '%I',
@@ -2138,7 +2149,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '12',
             fmt = '%I',
         },
-        -- %j
+        -- %j is replaced by the day of the year as a decimal
+        -- number (001-366).
         {
             buf = '001',
             fmt = '%j',
@@ -2147,7 +2159,9 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '31-12-2024 366',
             fmt = '%d-%m-%Y %j',
         },
-        -- %k
+        -- %k is replaced by the hour (24-hour clock) as
+        -- a decimal number (0-23); single digits are preceded
+        -- by a blank.
         {
             buf = ' 0',
             fmt = '%k',
@@ -2156,7 +2170,9 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '23',
             fmt = '%k',
         },
-        -- %l
+        -- %l is replaced by the hour (12-hour clock) as
+        -- a decimal number (1-12); single digits are preceded
+        -- by a blank.
         {
             buf = ' 1',
             fmt = '%l',
@@ -2165,7 +2181,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '12',
             fmt = '%l',
         },
-        -- %m
+        -- %m is replaced by the month as a decimal number
+        -- (01-12).
         {
             buf = '01',
             fmt = '%m',
@@ -2174,7 +2191,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '12',
             fmt = '%m',
         },
-        -- %M
+        -- %M is replaced by the minute as a decimal number
+        -- (00-59).
         {
             buf = '00',
             fmt = '%M',
@@ -2183,13 +2201,15 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '59',
             fmt = '%M',
         },
-        -- %n
+        -- %n is replaced by a newline.
         {
             buf = '\n',
             fmt = '%n',
         },
-        -- %O, modifier, skipped.
-        -- %p
+        -- %O the same as %E*, skipped.
+        -- %p is replaced by national representation of either
+        -- "ante meridiem" (a.m.) or "post meridiem" (p.m.) as
+        -- appropriate.
         {
             buf = 'PM',
             fmt = '%p',
@@ -2212,12 +2232,14 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '18:20',
             fmt = '%R',
         },
-        -- %s
+        -- %s is replaced by the number of seconds since the
+        -- Epoch, UTC (see mktime(3)).
         {
             buf = '26-08-2024 1724630400',
             fmt = '%d-%m-%Y %s',
         },
-        -- %S
+        -- %S is replaced by the second as a decimal number
+        -- (00-60).
         {
             buf = '00',
             fmt = '%S',
@@ -2226,7 +2248,7 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '59',
             fmt = '%S',
         },
-        -- %t
+        -- %t is replaced by a tab.
         {
             buf = '26-08-2024 \t',
             fmt = '%d-%m-%Y %t',
@@ -2240,8 +2262,9 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '23:59:59',
             fmt = '%T',
         },
-        -- %u
-        -- Works for 4 ('Thursday') only when date is not
+        -- %u is replaced by the weekday (Monday as the first day
+        -- of the week) as a decimal number (1-7).
+        -- Note, works for 4 ('Thursday') only when date is not
         -- complete.
         {
             buf = '4',
@@ -2255,7 +2278,9 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '26-08-2024 1',
             fmt = '%d-%m-%Y %u',
         },
-        -- %U
+        -- %U is replaced by the week number of the year (Sunday
+        -- as the first day of the week) as a decimal number
+        -- (00-53).
         {
             buf = '01-01-2024 00',
             fmt = '%d-%m-%Y %U',
@@ -2270,8 +2295,10 @@ test:test("Parse strptime format and check with :format()", function(test)
             fmt = '%v',
         },
         -- %V, unsupported by strptime(3).
-        -- %w
-        -- Works for 4 (Thursday) only when date is not complete.
+        -- %w is replaced by the weekday (Sunday as the first day
+        -- of the week) as a decimal number (0-6).
+        -- Note, works for 4 (Thursday) only when date is not
+        -- complete.
         {
             buf = '4',
             fmt = '%w',
@@ -2280,17 +2307,19 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '26-08-2024 1',
             fmt = '%d-%m-%Y %w',
         },
-        -- %W
+        -- %W is replaced by the week number of the year (Monday
+        -- as the first day of the week) as a decimal number
+        -- (00-53).
         {
             buf = '26-08-2024 35',
             fmt = '%d-%m-%Y %W',
         },
-        -- %x
+        -- %x is replaced by national representation of the date.
         {
             buf = '01/01/70',
             fmt = '%x',
         },
-        -- %X
+        -- %X is replaced by national representation of the time.
         {
             buf = '00:00:00',
             fmt = '%X',
@@ -2299,7 +2328,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '23:59:59',
             fmt = '%X',
         },
-        -- %y
+        -- %y is replaced by the year without century as a decimal
+        -- number (00-99).
         {
             buf = '00',
             fmt = '%y',
@@ -2308,7 +2338,8 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '99',
             fmt = '%y',
         },
-        -- %Y
+        -- %Y is replaced by the year with century as a decimal
+        -- number.
         {
             buf = '0000',
             fmt = '%Y',
@@ -2317,7 +2348,11 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '3000',
             fmt = '%Y',
         },
-        -- %z
+        -- %z is replaced by the time zone offset from UTC;
+        -- a leading plus sign stands for east of UTC, a minus
+        -- sign for west of UTC, hours and minutes follow with
+        -- two digits each and no delimiter between them
+        -- (common form for RFC 822 date headers).
         {
             buf = '+0403',
             fmt = '%z',
@@ -2326,17 +2361,19 @@ test:test("Parse strptime format and check with :format()", function(test)
             buf = '-0403',
             fmt = '%z',
         },
-        -- %Z
+        -- %Z is replaced by the time zone name.
         {
             buf = 'MSK',
             fmt = '%Z',
         },
-        -- %+
+        -- %+ is replaced by national representation of the date
+        -- and time (the format is similar to that produced by
+        -- date(1)).
         {
             buf = 'Mon Aug 26 00:00:00  2024',
             fmt = '%+',
         },
-        -- %%
+        -- %% is replaced by `%'.
         {
             buf = '26-08-2024 %',
             fmt = '%d-%m-%Y %%',
```